tests/app/celery/test_provider_tasks.py

```diff
@@ -180,3 +205,282 @@ def test_if_ses_send_rate_throttle_then_should_retry_and_log_warning(sample_noti
     assert sample_notification.status == "created"
     assert not mock_logger_exception.called
     assert mock_logger_warning.called
+
+
+@freeze_time("2020-02-17 16:00:00")
+def test_update_letter_to_sending(sample_letter_template):
+    letter = create_notification(
+        template=sample_letter_template,
+        status=NOTIFICATION_CREATED,
+        created_at=datetime.now(),
+    )
+    update_letter_to_sending(letter)
+
+    assert letter.status == NOTIFICATION_SENDING
+    assert letter.sent_at == datetime.utcnow()
+    assert letter.updated_at == datetime.utcnow()
+    assert letter.sent_by == "dvla"
+
+
+@mock_s3
+@freeze_time("2020-02-17 16:00:00")
+@pytest.mark.parametrize(
+    "is_precompiled, to_field, personalisation",
+    [
+        (
+            True,
+            "A. User\nMy Street,\nLondon,\nSW1 1AA",
+            {"address_line_1": "Provided as PDF"},
+        ),
+        (
+            False,
+            "A. User",
+            {
+                "addressline1": "A. User",
+                "addressline2": "My Street",
+                "addressline3": "London",
+                "addressline4": "SW1 1AA",
+                "addressline5": None,
+                "addressline6": None,
```

Comment:
Why the difference in snake casing between these two parameterised examples? PostalAddress seems to want them with underscores.

edit: omg I've just read what InsensitiveDict does 🤯😱 That name is not very accurate. I really don't like how forgiving this is for address info.

Reply:
I realised when writing this code the address keys are different for CSV vs one-off letters and the `InsensitiveDict` is used to normalise things. Also that the address details we store in the `to` and the personalisation field vary depending on how the letter was sent, so it's all very inconsistent unfortunately

Reply:
Oh no :(